packages/js/src/ui/context/LocalizationContext.tsx

```diff
@@ -20,18 +50,33 @@ const LocalizationContext = createContext<LocalizationContextType | undefined>(u
 type LocalizationProviderProps = ParentProps & { localization?: Localization };
 
 export const LocalizationProvider = (props: LocalizationProviderProps) => {
-  const localization = createMemo(() => ({ ...defaultLocalization, ...(props.localization || {}) }));
+  const splitLocalization = createMemo(() => {
+    const { dynamic, ...rest } = props.localization || {};
+
+    return { dynamic: dynamic || {}, localizationObject: rest };
+  });
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** is this separate memo needed? can't we do this code in `const localization = createMemo` ?